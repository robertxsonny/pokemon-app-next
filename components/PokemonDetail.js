import styled from '@emotion/styled';

import PokemonType from './PokemonType';
import PokemonMoves from './PokemonMoves';
import { SentenceCase } from '../styles/shared';
import { md, sm } from '../styles/breakpoints';

const DetailContent = styled.section({
  flex: 1,
  paddingBottom: 84
})

const DetailHeader = styled.div({
  backgroundColor: 'white',
  'h1, h3': {
    margin: 0
  },
  [sm.max]: {
    position: 'sticky',
    top: 0,
    padding: '12px 0'
  },
  [md.min]: {
    display: 'flex',
    alignItems: 'flex-end',
    justifyContent: 'flex-start'
  }
})

const Title = styled(SentenceCase.withComponent('h1'))({
  [sm.max]: {
    textAlign: 'center',
    fontSize: 24,
    'small': {
      marginLeft: 8,
      fontSize: 14
    }
  },
  [md.min]: {
    paddingRight: 16,
    'small': {
      marginLeft: 8,
      fontSize: 16
    }
  }
});

const CenteredFlex = styled.div({
  display: 'flex',
  alignItems: 'flex-end',
  justifyContent: 'center'
})

const SectionTitle = styled.h2({
  fontSize: 20,
  'small': {
    fontSize: 16,
    marginLeft: 6
  }
});

const SmallCardWrapper = styled(CenteredFlex)({
  margin: '16px -8px',
  [md.min]: {
    justifyContent: 'flex-start'
  }
})

const SmallCard = styled.div({
  backgroundColor: 'slategray',
  color: 'white',
  width: 120,
  height: 90,
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  borderRadius: 6,
  margin: 8,
  fontSize: 13,
  'h3': {
    marginTop: 0,
    marginBottom: 8,
    fontSize: 28
  },
  [sm.max]: {
    width: 80,
    height: 60,
    margin: 4,
    fontSize: 12,
    'h3': {
      marginBottom: 4,
      fontSize: 20
    }
  }
})


const PokemonDetail = ({ name, nickname, types, moves, height, weight, numOfCollected }) => (
  <DetailContent>
    <DetailHeader>
      <Title>
        {nickname ? (<>{nickname}<small>({name})</small></>) : name}
      </Title>
      <CenteredFlex>{types.map(({ type }) => <PokemonType key={type.name} type={type.name} />)}</CenteredFlex>
    </DetailHeader>
    <SmallCardWrapper>
      {height && (
        <SmallCard>
          <h3>{height}</h3>
          Height (in)
        </SmallCard>
      )}
      {weight && (
        <SmallCard>
          <h3>{weight}</h3>
          Weight (lbs)
        </SmallCard>
      )}
      {Number.isInteger(numOfCollected) && (
        <SmallCard>
          <h3>{numOfCollected}</h3>
          In Collection
        </SmallCard>
      )}
    </SmallCardWrapper>
    <div>
      <SectionTitle>Moves</SectionTitle>
      <PokemonMoves moves={moves} />
    </div>
  </DetailContent>
)

export default PokemonDetail;