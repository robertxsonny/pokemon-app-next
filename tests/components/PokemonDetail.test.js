import { render } from '@testing-library/react';
import PokemonDetail from '../../components/PokemonDetail';

const types = [{ type: { name: 'electric' } }]

const moves = [
  { move: { name: 'pound' } },
  { move: { name: 'karate-chop' } }
];

describe('PokemonDetailItem component', () => {
  test('Should render pokemon detail', () => {
    const { container } = render(<PokemonDetail name="Pikachu" types={types} moves={moves} />);
    expect(container).toMatchSnapshot();
  })

  test('Should render pokemon detail with nickname', () => {
    const { container } = render(<PokemonDetail nickname="Pikapika" name="Pikachu" types={types} moves={moves} />);
    expect(container).toMatchSnapshot();
  })

  test('Should render pokemon detail with height and weight', () => {
    const { container } = render(<PokemonDetail name="Pikachu" types={types} moves={moves} height={4} weight={600} />);
    expect(container).toMatchSnapshot();
  })

  test('Should render pokemon detail with number of collected pokemons', () => {
    const { container } = render(<PokemonDetail name="Pikachu" types={types} moves={moves} height={4} weight={600} numOfCollected={2} />);
    expect(container).toMatchSnapshot();
  })

})