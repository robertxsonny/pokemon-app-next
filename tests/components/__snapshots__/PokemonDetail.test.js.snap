// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetailItem component Should render pokemon detail 1`] = `
<div>
  <section
    class="css-56c5k9-DetailContent ert5fdg6"
  >
    <div
      class="css-m4xjt3-DetailHeader ert5fdg5"
    >
      <h1
        class="css-1buz0g5-Title-SentenceCase-Title ert5fdg4"
      >
        Pikachu
      </h1>
      <div
        class="css-8evzty-CenteredFlex ert5fdg3"
      >
        <div
          class="css-1r0vg09-TypeLabel-electric e5gu2840"
          type="electric"
        >
          electric
        </div>
      </div>
    </div>
    <div
      class="css-1gn2694-CenteredFlex-SmallCardWrapper ert5fdg1"
    />
    <div>
      <h2
        class="css-5n37sm-SectionTitle ert5fdg2"
      >
        Moves
      </h2>
      <div
        class="css-1oxeyy9-MoveItemsOuter e1mi334c3"
      >
        <div
          class="css-1nvhei2-MoveItemsWrapper e1mi334c2"
        >
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            pound
          </div>
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            karate chop
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`PokemonDetailItem component Should render pokemon detail with height and weight 1`] = `
<div>
  <section
    class="css-56c5k9-DetailContent ert5fdg6"
  >
    <div
      class="css-m4xjt3-DetailHeader ert5fdg5"
    >
      <h1
        class="css-1buz0g5-Title-SentenceCase-Title ert5fdg4"
      >
        Pikachu
      </h1>
      <div
        class="css-8evzty-CenteredFlex ert5fdg3"
      >
        <div
          class="css-1r0vg09-TypeLabel-electric e5gu2840"
          type="electric"
        >
          electric
        </div>
      </div>
    </div>
    <div
      class="css-1gn2694-CenteredFlex-SmallCardWrapper ert5fdg1"
    >
      <div
        class="css-2y01tc-SmallCard ert5fdg0"
      >
        <h3>
          4
        </h3>
        Height (in)
      </div>
      <div
        class="css-2y01tc-SmallCard ert5fdg0"
      >
        <h3>
          600
        </h3>
        Weight (lbs)
      </div>
    </div>
    <div>
      <h2
        class="css-5n37sm-SectionTitle ert5fdg2"
      >
        Moves
      </h2>
      <div
        class="css-1oxeyy9-MoveItemsOuter e1mi334c3"
      >
        <div
          class="css-1nvhei2-MoveItemsWrapper e1mi334c2"
        >
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            pound
          </div>
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            karate chop
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`PokemonDetailItem component Should render pokemon detail with nickname 1`] = `
<div>
  <section
    class="css-56c5k9-DetailContent ert5fdg6"
  >
    <div
      class="css-m4xjt3-DetailHeader ert5fdg5"
    >
      <h1
        class="css-1buz0g5-Title-SentenceCase-Title ert5fdg4"
      >
        Pikapika
        <small>
          (
          Pikachu
          )
        </small>
      </h1>
      <div
        class="css-8evzty-CenteredFlex ert5fdg3"
      >
        <div
          class="css-1r0vg09-TypeLabel-electric e5gu2840"
          type="electric"
        >
          electric
        </div>
      </div>
    </div>
    <div
      class="css-1gn2694-CenteredFlex-SmallCardWrapper ert5fdg1"
    />
    <div>
      <h2
        class="css-5n37sm-SectionTitle ert5fdg2"
      >
        Moves
      </h2>
      <div
        class="css-1oxeyy9-MoveItemsOuter e1mi334c3"
      >
        <div
          class="css-1nvhei2-MoveItemsWrapper e1mi334c2"
        >
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            pound
          </div>
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            karate chop
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`PokemonDetailItem component Should render pokemon detail with number of collected pokemons 1`] = `
<div>
  <section
    class="css-56c5k9-DetailContent ert5fdg6"
  >
    <div
      class="css-m4xjt3-DetailHeader ert5fdg5"
    >
      <h1
        class="css-1buz0g5-Title-SentenceCase-Title ert5fdg4"
      >
        Pikachu
      </h1>
      <div
        class="css-8evzty-CenteredFlex ert5fdg3"
      >
        <div
          class="css-1r0vg09-TypeLabel-electric e5gu2840"
          type="electric"
        >
          electric
        </div>
      </div>
    </div>
    <div
      class="css-1gn2694-CenteredFlex-SmallCardWrapper ert5fdg1"
    >
      <div
        class="css-2y01tc-SmallCard ert5fdg0"
      >
        <h3>
          4
        </h3>
        Height (in)
      </div>
      <div
        class="css-2y01tc-SmallCard ert5fdg0"
      >
        <h3>
          600
        </h3>
        Weight (lbs)
      </div>
      <div
        class="css-2y01tc-SmallCard ert5fdg0"
      >
        <h3>
          2
        </h3>
        In Collection
      </div>
    </div>
    <div>
      <h2
        class="css-5n37sm-SectionTitle ert5fdg2"
      >
        Moves
      </h2>
      <div
        class="css-1oxeyy9-MoveItemsOuter e1mi334c3"
      >
        <div
          class="css-1nvhei2-MoveItemsWrapper e1mi334c2"
        >
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            pound
          </div>
          <div
            class="css-1ewczkj-MoveItem e1mi334c1"
          >
            karate chop
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
